cf-net/cf-net.c

```diff
@@ -740,9 +768,62 @@ static int CFNetGet(ARG_UNUSED CFNetOptions *opts, const char *hostname, char **
         return -1;
     }
 
-    int ret = CFNetGetFile(hostname, remote_file, local_file);
+    CFNetThreadData **threads = (CFNetThreadData**) xcalloc(n_threads, sizeof(CFNetThreadData*));
+    for (int i = 0; i < n_threads; i++)
+    {
+        threads[i] = (CFNetThreadData*) xcalloc(1, sizeof(CFNetThreadData));
+        threads[i]->data = (GetFileData*) xcalloc(1, sizeof(GetFileData));
+        threads[i]->data->hostname = hostname;
+        threads[i]->data->remote_file = remote_file;
+        if (n_threads > 1)
+        {
+            if (strstr(local_file, "%d") != NULL)
+            {
+                snprintf(threads[i]->data->local_file, PATH_MAX, local_file, i);
```

Comment:
dangerous but I guess it's ok for a testing tool

Reply:
Yeah, there's practically no chance that ``local_file`` will not be ``\0``-terminated.

Reply:
it's not for the NULL termination, it's for the possibility of putting format string qualifiers in the filename. Imagine a filename "blah%s%x%d".

Reply:
Oh, that's right. Well, we are all responsible people not doing things like this. :)